libraries/bsg_manycore.cpp

```diff
@@ -1375,22 +1376,25 @@ static int hb_mc_manycore_write(hb_mc_manycore_t *mc, const hb_mc_npa_t *npa, co
         if (err != HB_MC_SUCCESS)
                 return err;
 
-        int shift = hb_mc_npa_get_epa(npa) & 0x3;
+        hb_mc_epa_t epa = hb_mc_npa_get_epa(npa);
+
+        // Check for 4-byte, 2-byte or 1-byte alignment based on size
+        if (hb_mc_epa_check_alignment(&epa, sz) != HB_MC_SUCCESS)
+                return HB_MC_INVALID;
+
+        int shift = epa & 0x3;
         /* set data and size */
         switch (sz) {
         case 4:
```

Comment:
What if we define HB_MC_SIZE_32/16/8 (or similar). Naming doesn't matter, but removing the last magic numbers here to make it more obvious would be nice

Reply:
Personally, I don't really see the point. There's 0% chance we would ever change the values of those constants. It's almost like having a macro like this:
```
#define ONE 1
```
I also don't think it makes it more legible.

Reply:
#define SGTM 1